core/xml.js

```diff
@@ -471,6 +478,24 @@ Blockly.Xml.domToBlock = function(xmlBlock, workspace) {
 };
 
 /**
+ * Decode an XML list of variables and add the variables to the workspace.
+ * @param {!Element} xmlVariables List of XML variable elements.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ */
+Blockly.Xml.domToVariables = function(xmlVariables, workspace) {
+  for (var i = 0, xmlChild; xmlChild = xmlVariables.children[i]; i++) {
+    var type = xmlChild.getAttribute('type');
+    var id = xmlChild.getAttribute('id');
+    var name = xmlChild.textContent;
+
+    if (!type && type != '') {
```

Comment:
Oh, nm. I read it wrong. Maybe make this more explicit? Something like if (type == null) or if (typeof(type) === "undefined" || type === null)? @rachel-fenichel if there's a more standard js way of doing this.